src/lib.rs

```diff
@@ -1,334 +1,201 @@
+                if e.file_type().expect("File broken").is_dir() {
+                    None
+                } else {
+                    Some(format!("{}/{}", proto_dir, e.file_name().to_string_lossy()))
                 }
-            }
-
-            panic!("file {:?} is not found in includes {:?}", file, includes);
-        }
-
-        protobuf_codegen::gen_and_write(
-            desc.get_file(),
-            &files_to_generate,
-            &Path::new(out_dir),
-            &protobuf_codegen::Customize::default(),
-        )
-        .unwrap();
-        generate_grpcio(&desc.get_file(), &files_to_generate, out_dir);
-        replace_read_unknown_fields(out_dir);
+            })
+            .collect();
+        self
     }
 
-    #[cfg(feature = "grpcio-protobuf-codec")]
-    pub fn generate_grpcio(
-        desc: &[protobuf::descriptor::FileDescriptorProto],
-        files_to_generate: &[String],
-        out_dir: &str,
-    ) {
-        use std::io::Write;
-
-        let output_dir = std::path::Path::new(out_dir);
-        let results = grpcio_compiler::codegen::gen(desc, &files_to_generate);
-        for res in results {
-            let out_file = output_dir.join(&res.name);
-            let mut f = std::fs::File::create(&out_file).unwrap();
-            f.write_all(&res.content).unwrap();
-        }
+    pub fn files<T: Into<String> + Clone>(&mut self, files: &[T]) -> &mut Self {
+        self.files = files
+            .iter()
+            .map(|t| t.clone().into())
+            .collect::<Vec<String>>();
+        self
     }
 
-    #[cfg(all(feature = "protobuf-codec", not(feature = "grpcio-protobuf-codec")))]
-    pub fn generate_grpcio(_: &[protobuf::descriptor::FileDescriptorProto], _: &[String], _: &str) {
+    pub fn includes(&mut self, includes: Vec<String>) -> &mut Self {
+        self.includes = includes;
+        self
     }
 
-    /// Convert protobuf files to use the old way of reading protobuf enums.
-    // FIXME: Remove this once stepancheg/rust-protobuf#233 is resolved.
-    pub fn replace_read_unknown_fields(out_dir: &str) {
-        let regex =
-            Regex::new(r"::protobuf::rt::read_proto3_enum_with_unknown_fields_into\(([^,]+), ([^,]+), &mut ([^,]+), [^\)]+\)\?").unwrap();
-        for f in fs::read_dir(out_dir).unwrap() {
-            let path = match f {
-                Ok(p) => p.path(),
-                Err(e) => panic!("failed to list {}: {:?}", out_dir, e),
-            };
-            if path.extension() != Some(std::ffi::OsStr::new("rs")) {
-                continue;
-            }
+    pub fn append_include(&mut self, include: String) -> &mut Self {
+        self.includes.push(include);
+        self
+    }
 
-            let mut text = String::new();
-            let mut f = File::open(&path).unwrap();
-            f.read_to_string(&mut text)
-                .expect("Couldn't read source file");
+    pub fn include_black_list(&mut self, include_black_list: Vec<String>) -> &mut Self {
```

Comment:
In my opinion, we barely set them all at a time as there will be double allocations.

Reply:
So do you prefer to have just a method to append and a method to clear?